browser/ui/webui/new_tab_page/brave_new_tab_page_handler.cc

```diff
@@ -223,28 +263,65 @@ void BraveNewTabPageHandler::OnSavedCustomImage(const base::FilePath& path) {
     return;
   }
 
-#if defined(OS_WIN)
-  auto file_name = base::SysWideToUTF8(path.BaseName().value());
-#else
-  auto file_name = std::string(path.BaseName().value().c_str());
-#endif
+  if (brave_new_tab_page::mojom::kMaxCustomImageBackgrounds -
+          NTPBackgroundPrefs(profile_->GetPrefs())
+              .GetCustomImageList()
+              .size() <=
+      0) {
+    // We can't save more images.
+    file_manager_->RemoveImage(path, base::DoNothing());
+    return;
+  }
+
+  auto file_name =
+      CustomBackgroundFileManager::Converter(path).To<std::string>();
   DCHECK(!file_name.empty());
 
   auto background_pref = NTPBackgroundPrefs(profile_->GetPrefs());
   background_pref.SetType(NTPBackgroundPrefs::Type::kCustomImage);
   background_pref.SetSelectedValue(file_name);
   background_pref.AddCustomImageToList(file_name);
   OnBackgroundUpdated();
+  OnCustomImageBackgroundsUpdated();
+}
+
+void BraveNewTabPageHandler::OnRemoveCustomImageBackground(
+    const base::FilePath& path,
+    bool success) {
+  if (!success) {
+    LOG(ERROR) << "Failed to remove custom image " << path;
+    return;
+  }
+
+  auto file_name =
+      CustomBackgroundFileManager::Converter(path).To<std::string>();
+  DCHECK(!file_name.empty());
+
+  auto background_pref = NTPBackgroundPrefs(profile_->GetPrefs());
+  background_pref.RemoveCustomImageFromList(file_name);
+  if (background_pref.GetType() == NTPBackgroundPrefs::Type::kCustomImage &&
```

Comment:
If there are another custom images, I think one of them could be selected instead of going to brave option. WDYT?

Reply:
Yeah, that's good idea. Added fallback.